packages/core/core/src/services/entity-validator/validators.ts

```diff
@@ -149,14 +149,11 @@ const addUniqueValidator = <T extends strapiUtils.yup.AnySchema>(
   return validator.test('unique', 'This attribute must be unique', async (value) => {
     const isPublish = options.isDraft === false;
 
-    // When publishing, the value will not have changed so we take the value from the entity
-    const valueToCheck = isPublish ? entity?.[updatedAttribute.name] : value;
-
```

Comment:
@jhoward1994 do you think removing this line could give us problems?
Even if the value has not changed, now the `value` does contain the unique value to publish, and `entity?.[updatedAttribute.name]` does not.

Reply:
all tests are passing in my machine, don't know why are they failing here 🤔

Reply:
I think the logic of this validator should follow https://github.com/strapi/strapi/pull/19549/files#r1495557301

e.g. take the value provided to the validator (as you are here) or default to the value on the existing entity. In that PR this fixes the validation for cloning

Do you see any problems with that?

Reply:
that will probably not work when the value to update is null,
if trying to set a value to null, the check will ignore the new value and work on the previous one

Reply:
Also! if I understand correctly `entity?.[updatedAttribute.name]` should be the previous value of an entity, so does it make sense to check if the previous value is unique or not? shouldn't already be unique if it's already in db?